pandas/core/internals/array_manager.py

```diff
@@ -515,13 +516,19 @@ def convert(
         numeric: bool = True,
         timedelta: bool = True,
     ) -> ArrayManager:
-        return self.apply_with_block(
-            "convert",
-            copy=copy,
-            datetime=datetime,
-            numeric=numeric,
-            timedelta=timedelta,
-        )
+        def _convert(arr):
```

Comment:
not pretty but ok

Reply:
What's not pretty about it? The inline function?

Reply:
soft_convert_objects can just do this directly right? (or the copy keyword is not functioning)

Reply:
Currently `soft_convert_objects` assumes the input is object dtype, and doesn't do such checking. I assume the check could be moved into ``soft_convert_objects`` as well (although eg for Block, which has an ObjectBlock, it knows it is ndarray[object], and such an additional check would be unnecessary)

Reply:
BTW soft_convert_objects is only used for ObjectBlock.convert ATM, so no real harm in making it handle non-object with these extra two lines (though to get effective simplification in the BM code might need to tweak copy kwarg behavior)

Reply:
ok if would revisit / consolidate logic at some point here.